### PR TITLE
docs(extract): add extraction rules reference, metadata enrichment, and AI scanner prompt

### LINKED ISSUE #170: [M6] Create extraction documentation and AI scanner prompt
### Context
**PRD:** `docs/project/PRD/active/PRD-phase-11-metadata-extraction.md` — Milestone 6

Documentation enables teams to adopt metadata extraction. The AI scanner prompt accelerates onboarding by analyzing codebases and recommending configs.

### Traceability
- **PRD Section:** M6-D6.1 (Workflow docs), M6-D6.2 (Config reference), M6-D6.3 (AI scanner), M6-D6.4 (Architecture)
- **Success Criteria:** SC-8 (AI scanner prompt exists), SC-9 (Workflow docs updated), SC-10 (Config reference complete)

### Acceptance Criteria
- [ ] `docs/workflow/` includes metadata extraction step with before/after examples
- [ ] `docs/reference/extraction-config.md` documents all 15 extraction rules with examples
- [ ] Each rule has: description, YAML syntax, TypeScript example, edge cases
- [ ] AI codebase scanner prompt at `docs/ai-prompts/codebase-scanner.md`
- [ ] AI prompt outputs: extraction config, ESLint config snippet, architectural test suggestions
- [ ] AI prompt tested on ecommerce-demo-app (follows conventions)
- [ ] AI prompt tested on non-convention codebase (detection works on inconsistent code)
- [ ] Generated config validates against JSON schema
- [ ] Generated config produces non-empty extraction results
- [ ] Generated ESLint config snippet is syntactically valid
- [ ] Architecture overview at `docs/architecture/overview.md` reflects metadata extraction capability

### Edge Case Scenarios
- Codebase with decorators → AI recommends decorator-based detection
- Codebase with naming conventions → AI recommends nameEndsWith/nameMatches
- Codebase with interfaces → AI recommends implementsInterface
- Mixed conventions (< 50% follow majority) → AI suggests migration path
- No clear conventions → AI suggests starting with recommended interfaces

### Implementation Guidelines
- **Workflow docs:** `docs/workflow/metadata-extraction.md`
- **Config reference:** `docs/reference/extraction-config.md`
- **AI prompt:** `docs/ai-prompts/codebase-scanner.md`
- **Architecture:** Update `docs/architecture/overview.md`
- Config reference should be exhaustive (all rules, all options)
- AI prompt should be self-contained (works with any LLM)
- Include validation criteria in AI prompt (schema validation, non-empty output)

### Embedded Reasoning
**Why:** Documentation reduces adoption friction. AI scanner automates config generation.
**What:** Workflow guide, comprehensive config reference, AI prompt for codebase analysis.
**How:** Markdown docs, structured AI prompt with examples and validation criteria.

### Testing Strategy
- **Manual:** Review docs for completeness and accuracy
- **Validation:** AI prompt tested on 2 codebases (convention-following, non-convention)
- **Validation:** Generated configs validated against schema
- **Integration:** Follow workflow docs to extract from sample codebase

### Dependencies
Depends on #166 (CLI flags documented)

### Verification
```bash
# Docs exist
ls docs/workflow/metadata-extraction.md
ls docs/reference/extraction-config.md
ls docs/ai-prompts/codebase-scanner.md
# AI prompt produces valid config (manual test)
# Generated config validates:
# riviere extract --config ./generated-config.yaml --dry-run
```

---

## Amendment

Acceptance criteria says '15 extraction rules' but only 11 exist in the codebase (types.ts ExtractionRule union has 11 members). Updating criteria to reflect actual count of 11. Also, user directed that docs go in apps/docs/ (public site) not docs/workflow/ or docs/reference/ (internal), so file paths differ from original criteria.

### PR DESCRIPTION
**Pull request by NTCoding**
Closes #170

### Context
**PRD:** `docs/project/PRD/active/PRD-phase-11-metadata-extraction.md` — Milestone 6

Documentation enables teams to adopt metadata extraction. The AI scanner prompt accelerates onboarding by analyzing codebases and recommending configs.

### Traceability
- **PRD Section:** M6-D6.1 (Workflow docs), M6-D6.2 (Config reference), M6-D6.3 (AI scanner), M6-D6.4 (Architecture)
- **Success Criteria:** SC-8 (AI scanner prompt exists), SC-9 (Workflow docs updated), SC-10 (Config reference complete)

### Acceptance Criteria
- [ ] `docs/workflow/` includes metadata extraction step with before/after examples
- [ ] `docs/reference/extraction-config.md` documents all 15 extraction rules with examples
- [ ] Each rule has: description, YAML syntax, TypeScript example, edge cases
- [ ] AI codebase scanner prompt at `docs/ai-prompts/codebase-scanner.md`
- [ ] AI prompt outputs: extraction config, ESLint config snippet, architectural test suggestions
- [ ] AI prompt tested on ecommerce-demo-app (follows conventions)
- [ ] AI prompt tested on non-convention codebase (detection works on inconsistent code)
- [ ] Generated config validates against JSON schema
- [ ] Generated config produces non-empty extraction results
- [ ] Generated ESLint config snippet is syntactically valid
- [ ] Architecture overview at `docs/architecture/overview.md` reflects metadata extraction capability

### Edge Case Scenarios
- Codebase with decorators → AI recommends decorator-based detection
- Codebase with naming conventions → AI recommends nameEndsWith/nameMatches
- Codebase with interfaces → AI recommends implementsInterface
- Mixed conventions (< 50% follow majority) → AI suggests migration path
- No clear conventions → AI suggests starting with recommended interfaces

### Implementation Guidelines
- **Workflow docs:** `docs/workflow/metadata-extraction.md`
- **Config reference:** `docs/reference/extraction-config.md`
- **AI prompt:** `docs/ai-prompts/codebase-scanner.md`
- **Architecture:** Update `docs/architecture/overview.md`
- Config reference should be exhaustive (all rules, all options)
- AI prompt should be self-contained (works with any LLM)
- Include validation criteria in AI prompt (schema validation, non-empty output)

### Embedded Reasoning
**Why:** Documentation reduces adoption friction. AI scanner automates config generation.
**What:** Workflow guide, comprehensive config reference, AI prompt for codebase analysis.
**How:** Markdown docs, structured AI prompt with examples and validation criteria.

### Testing Strategy
- **Manual:** Review docs for completeness and accuracy
- **Validation:** AI prompt tested on 2 codebases (convention-following, non-convention)
- **Validation:** Generated configs validated against schema
- **Integration:** Follow workflow docs to extract from sample codebase

### Dependencies
Depends on #166 (CLI flags documented)

### Verification
```bash
# Docs exist
ls docs/workflow/metadata-extraction.md
ls docs/reference/extraction-config.md
ls docs/ai-prompts/codebase-scanner.md
# AI prompt produces valid config (manual test)
# Generated config validates:
# riviere extract --config ./generated-config.yaml --dry-run
```

---

## Amendment

Acceptance criteria says '15 extraction rules' but only 11 exist in the codebase (types.ts ExtractionRule union has 11 members). Updating criteria to reflect actual count of 11. Also, user directed that docs go in apps/docs/ (public site) not docs/workflow/ or docs/reference/ (internal), so file paths differ from original criteria.

## Summary by CodeRabbit

* **New Features**
  * Added CLI options to control output file, enrichment input, base branch, file selection, output format, and flags for components-only, allow-incomplete, and PR-based extraction.

* **Documentation**
  * Published a new Extraction Rules reference.
  * Updated extraction workflow navigation to AI-assisted paths.
  * Added metadata enrichment guidance with examples.
  * Introduced comprehensive documentation governance and review guidance.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: apps/docs/reference/cli/cli-reference.md [hunk 1]
```diff
@@ -691,11 +691,19 @@ riviere extract [options]
 | Flag | Description |
 |------|-------------|
 | `--config <path>` | Path to extraction config file |
+| `-o, --output <file>` | Write output to file instead of stdout |
+| `--enrich <file>` | Read draft components from file and enrich with extraction rules |
+| `--base <branch>` | Override base branch for --pr (default: auto-detect) |
+| `--files <paths...>` | Extract from specific files |
+| `--format <type>` | Output format: json (default) or markdown |
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply apps/docs/reference/cli/cli-reference.md]: ❌ **Rejected**: cli-reference.md is auto-generated by pnpm nx generate-docs riviere-cli. The flag categorization comes from the CLI command definitions. If the categorization is wrong, it needs to be fixed in the source command definitions, not in the generated docs.

---
*Generated by Claude Code*

File: .claude/commands/documentation-review.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+```text
+Read apps/docs/CLAUDE.md for the page format rules.
+
+For ALL changed docs files: [list paths]
+
+For each file, determine its type (reference, workflow step, overview/index)
+and compare against the canonical format in CLAUDE.md:
+- Reference pages: compare against apps/docs/reference/extraction-config/predicates.md
+- Workflow pages: compare against apps/docs/extract/ai-assisted/step-1-understand.md
+- Overview pages: compare against apps/docs/extract/index.md
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply .claude/commands/documentation-review.md]: ✅ **Fixed**: Updated canonical path to apps/docs/extract/deterministic/typescript/workflow/step-1-understand.md — the ai-assisted steps are auto-generated from CLI package at build time, not hand-written.

---
*Generated by Claude Code*

File: package.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
...
     "build": "nx run-many -t build",
-    "build:deploy": "cd packages/riviere-schema && npx tsc --build tsconfig.lib.json && cd ../riviere-query && npx tsc --build tsconfig.lib.json && cd ../../apps/docs && cp ../../packages/riviere-cli/docs/workflow/step-*.md extract/steps/ && cp ../../packages/riviere-cli/docs/generated/cli-reference.md reference/cli/cli-reference.md && NODE_OPTIONS='--max-old-space-size=4096' npx vitepress build && cd ../eclair && npx vite build && cd ../.. && rm -rf dist && mkdir -p dist/eclair && cp -r apps/docs/.vitepress/dist/* dist/ && cp -r apps/eclair/dist/* dist/eclair/",
+    "build:deploy": "cd packages/riviere-schema && npx tsc --build tsconfig.lib.json && cd ../riviere-query && npx tsc --build tsconfig.lib.json && cd ../../apps/docs && cp ../../packages/riviere-cli/docs/workflow/step-*.md extract/ai-assisted/ && cp ../../packages/riviere-cli/docs/generated/cli-reference.md reference/cli/cli-reference.md && NODE_OPTIONS='--max-old-space-size=4096' npx vitepress build && cd ../eclair && npx vite build && cd ../.. && rm -rf dist && mkdir -p dist/eclair && cp -r apps/docs/.vitepress/dist/* dist/ && cp -r apps/eclair/dist/* dist/eclair/",
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply package.json]: ❌ **Rejected**: build:deploy is a deployment-specific assembly script that copies files, builds VitePress, and assembles the dist folder. It intentionally uses direct tool invocations because the deployment assembly has different requirements than dev builds. Refactoring this is out of scope for a documentation PR.

---
*Generated by Claude Code*
